plugins/secrets/vault/opts.go

```diff
@@ -24,3 +28,21 @@ func WithRenewal(d time.Duration) Opts {
 		v.renew = d
 	}
 }
+
+func WithKubernetesAuth() Opts {
+	return func(v *vault) {
+		addr := os.Getenv("VAULT_ADDR")
+		role := os.Getenv("DRONE_VAULT_KUBERNETES_ROLE")
+		mount := os.Getenv("DRONE_VAULT_AUTH_MOUNT_POINT")
+		jwtFile := "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
unsure the best place to put this. From my knowledge the location of this file will always be the same for Kubernetes pods.